### PR TITLE
Add new-parens rule

### LINKED ISSUE #1177: New Rule: new calls require parens
As seen here http://stackoverflow.com/a/3034952/3567713 brackets behind a new call are optional.
There should be a rule which require brackets.

### PR DESCRIPTION
**Pull request by YuichiNukiyama**
resolve #1177

### CODE REVIEW

File: test/rules/new-parens/test.ts.lint [hunk 1]
```diff
@@ -0,0 +1,9 @@
+// correct case
+var foo = new Date();
+
+// incorrect case
+var foge = new Date;
+           ~~~~~~~~ [new calls require parens]
+class Bar {}
+var bar = new Bar;
+          ~~~~~~~ [new calls require parens]
```
Comment (JKillian) [inline test/rules/new-parens/test.ts.lint]: Can we add one test case for something like `new Bar(new Foo)`?

File: src/rules/newParensRule.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+        return this.applyWithWalker(noVarKeywordWalker);
+    }
+}
+
+class NewParensWalker extends Lint.RuleWalker {
+    public visitNewExpression(node: ts.NewExpression) {
+        if (node.arguments === undefined) {
+            this.addFailure(this.createFailure(node.getStart(), node.getWidth(), Rule.FAILURE_STRING));
+        }
+        this.walkChildren(node);
```
Comment (JKillian) [inline src/rules/newParensRule.ts]: Just call `super.visitNewExpression(node)`

File: src/rules/newParensRule.ts [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+ */
+
+import * as ts from "typescript";
+import * as Lint from "../lint";
+
+export class Rule extends Lint.Rules.AbstractRule {
+    public static FAILURE_STRING = "new calls require parens";
+
+    public apply(sourceFile: ts.SourceFile): Lint.RuleFailure[] {
+        const noVarKeywordWalker = new NewParensWalker(sourceFile, this.getOptions());
```
Comment (JKillian) [inline src/rules/newParensRule.ts]: rename to `newParensWalker`

File: src/rules/newParensRule.ts [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as ts from "typescript";
+import * as Lint from "../lint";
+
+export class Rule extends Lint.Rules.AbstractRule {
+    public static FAILURE_STRING = "new calls require parens";
```
Comment (JKillian) [inline src/rules/newParensRule.ts]: can we make this "parentheses required when invoking a constructor"

**Comment by JKillian** [comment]
Thanks @YuichiNukiyama! Just a few small changes to make if you can

**Comment by YuichiNukiyama** [comment]
@JKillian I did it.

**Comment by JKillian** [comment]
Looks good! @adidahiya are you happy with mirroring the eslint terminology and calling this rule `new-parens`? Or should we go with something else?

**Comment by adidahiya** [comment]
`new-parens` sounds good to me

**Pull request merged**